modes/speedbar/evil-collection-speedbar.el

```diff
@@ -0,0 +1,105 @@
+;;; evil-collection-speedbar.el --- Evil bindings for speedbar -*- lexical-binding: t -*-
+
+;; Copyright (C) 2020 Zhiwei Chen
+
+;; Author: Zhiwei Chen <condy091@gmail.com>
+;; Maintainer: James Nguyen <james@jojojames.com>
+;; Pierre Neidhardt <mail@ambrevar.xyz>
+;; URL: https://github.com/emacs-evil/evil-collection
+;; Version: 0.0.1
+;; Package-Requires: ((emacs "25.1"))
+;; Keywords: evil, speedbar, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; Evil bindings for speedbar.
+
+;;; Code:
+
+(require 'evil-collection)
+(require 'speedbar)
+
+(defconst evil-collection-speedbar-maps '(speedbar-mode-map
+                                          speedbar-file-key-map
+                                          speedbar-buffers-key-map))
+
+;;;###autoload
+(defun evil-collection-speedbar-setup ()
+  "Set up `evil' bindings for `speedbar'."
+  (evil-set-initial-state 'speedbar-mode 'normal)
+  (evil-collection-define-key 'normal 'speedbar-mode-map
+    "q" 'dframe-close-frame
+    "j" 'speedbar-next
+    "k" 'speedbar-prev
+    [mouse-2] 'dframe-click
+    [follow-link] 'mouse-face
+    "gr" 'speedbar-refresh
+    "gg" 'evil-goto-first-line
```

Comment:
Do we need to bind these? (OK if so)

Reply:
No `gg` binding will result in `g g is undefined`, so bind it manually.
Binding `G` is for consistency. :-)